build2cmake/src/templates/preamble.cmake

```diff
@@ -1,6 +1,71 @@
 cmake_minimum_required(VERSION 3.26)
 project({{name}} LANGUAGES CXX)
 
+
+# Low-memory friendly defaults
+
+# Switch on automatically when the caller did **not** ask for a build type
+# (i.e. developer didn’t override with -DCMAKE_BUILD_TYPE=Debug or such)
+if(NOT CMAKE_BUILD_TYPE)
+  set(LOW_MEM ON)
+endif()
```

Comment:
I think we should only enable this if the user explicitly asks for it, not something that is implicitly enabled through another option.